controllers/front/ajax.php

```diff
@@ -42,14 +43,23 @@ private function submitHostedFields()
             if (!Order::getOrderByCartId($this->context->cart->id)) {
                 $this->validateOrder();
             }
+
+            /** @var SaferPayCardAliasRepository $cardAliasRep */
+            $cardAliasRep = $this->module->getModuleContainer()->get(SaferPayCardAliasRepository::class);
+
+            $selectedCard = Tools::getValue('selectedCard');
+
+            $alias = $cardAliasRep->getSavedCardAliasFromId($selectedCard);
+
             /** @var SaferPayInitialize $initializeService */
             $initializeService = $this->module->getModuleContainer()->get(SaferPayInitialize::class);
             $initializeBody = $initializeService->initialize(
                 Tools::getValue('paymentMethod'),
                 (int) Tools::getValue(SaferPayConfig::IS_BUSINESS_LICENCE),
-                -1,
-                null,
-                Tools::getValue('fieldToken')
+                $selectedCard,
+                $alias,
+                Tools::getValue('fieldToken'),
+                'successHosted'
```

Comment:
i would prefer const here that could explain what this string means, but just option